### DISCUSSION TITLE
Unable to Connect to RTSP stream

### DISCUSSION THREAD

**Discussion by bjs-pdx** [gtxaspec/wz_mini_hacks#236] [Q&A] [answered]
Trying to connect to two different cameras (PanV2 and regular V2) RTSP stream using VLC.  Keep getting a failure to connect message no matter what I try.  Seems like configuration of the cameras should be pretty straight forward and work without issues.  What am I doing wrong?  Trying to access via this string:  rtsp://admin:test1@192.168.x.x:8554/unicast (tried the other variations as well) but can never get a connection.  Here is the configuration of one of the cameras (ignore the RTSP_AUTH_DISABLE setting, I've tried it both ways).  The error from VLC is connection refused.

![Screenshot 2022-08-10 at 07-52-32 Wyze V3 on 4 49 1 47 running wz_mini Sun Aug 7 07 01 19 PM PDT 2022 as PanV2](https://user-images.githubusercontent.com/14082937/183937702-a215eac6-1cf9-44e0-aa1a-ebbb4b2b5a4f.png)

**Comment by haderp** [comment]
Are you using the VLC package installed through Debian packages by chance?  If so the Debian VLC package does not contain the RTSP playback libraries included.  More info here [https://forum.videolan.org/viewtopic.php?t=157795](https://forum.videolan.org/viewtopic.php?t=157795).  I know this because I spent 2 hours trying to get RTSP streaming to work and finally moved to a different OS and tested and it worked instantly.

**Reply by bjs-pdx** [reply]
It's not a vlc problem.  Works with the Pan V2 and the regular V2 but other apps cannot connect to the V2.

**Comment by Bob-AOL** [comment]
From the readme... (I initially had the same problem.)

-- the singular stream will be located at ```rtsp://login:password@IP_ADDRESS:8554/unicast```
**_multiple streams_** are located at ```rtsp://login:password@IP_ADDRESS:8554/video1_unicast``` and /video2_unicast``` respectively. --

Either disable the sub-stream for the single stream unicast only url, or specify the video sources accordingly  for using both streams.

**Reply by bjs-pdx** [reply]
I've tried all that.  Single stream, multiple stream, auth disabled, still get connection refused.

**Comment by gtxaspec** [comment]
If you are using the latest release, in the logs folder, do you see v4l2rtspserver.log?

**Reply by bjs-pdx** [reply]
Can't seem to find the logs folder.  

```
ls
backupa   configs   lib       opt       run       thirdlib
backupd   dev       linuxrc   params    sbin      tmp
backupk   driver    media     proc      sys       usr
bin       etc       mnt       root      system    var

```
Found /var/log but no v4l2rtspserver.log.

**Reply by baudneo** [reply]
I think it's /opt/wz_mini/logs

Try issuing `find / -iname '*.log'`

**Reply by bjs-pdx** [reply]
[log_gather_2022-08-10_T1936.tar.gz](https://github.com/gtxaspec/wz_mini_hacks/files/9303677/log_gather_2022-08-10_T1936.tar.gz)
Found the file there, but it is blank.  RTSP not starting?

The wz_init.log shows RTSP as disabled:

```
grep rtsp wz_init.log
#####S15v4l2rtspserver#####
rtsp disabled
rtsp disabled
sh: /opt/wz_mini/log/v4l2rtspserver.log: out of range
```

**Reply by baudneo** [reply]
Upload the gatherhxxxxx.tar.gz file that gather_wz_logs.sh generated.

**Reply by bjs-pdx** [reply]
See above.

**Reply by gtxaspec** [reply]
can you share the config, wz_mini.conf?

**Reply by bjs-pdx** [reply]
I re-did some changes to the wz_mini.conf and now both cameras are working.  I changed the user and password plus disabled auth.  Not sure what made it all start working.  I hate mysteries.  I did power cycle both cameras.  The V2 lost all the information in the config files (original and all backups) so I copied a manually edited one to the SD card and rebooted then everything worked.

**Reply by baudneo** [reply]
Guessing a typo or something similar?

**Reply by bjs-pdx** [reply]
Could be, but I'm USUALLY more careful than that.  Going to dig deeper, but happy it is working now.

**Reply by bjs-pdx** [reply]
Getting frustrated.  Got RTSP working fully on the PanV2.  I can connect to it from VLC and MotionEye.  However the V2 ONLY works with VLC.  When I try to add the camera to MotionEye I get connection refused using the same settings I did for VLC, while VLC is running the stream.  In both instances I use rtsp://192.168.5.105:8554/video7_unicast as the URL.  MotionEye has a separate field for the user and password.  Which log would show the attempted connection for RTSP?  Annoying that one camera works flawlessly and the other is a PITA.

**Reply by baudneo** [reply]
Check the motion eye logs, turn on debug logging. It may because the rtsp stream SDP params are not set correctly. I had issues using the rtsp streams with some software due to the profile-id being incorrect. 

Motioneye debug logs should tell you or give you a hint of what the issue may be.

**Reply by gtxaspec** [reply]
the panv2 is a t31 device, so the URL would be ` rtsp://192.168.5.105:8554/video2_unicast`

and logs for the rtsp server would be `/opt/wz_mini/log/v4l2rtspserver.log`

**Reply by bjs-pdx** [reply]
Yes, Pan V2 is working but the older V2 is not.  Tried both video6_unicast and video7_unicast and neither worked, nor did simply unicast.  I'll check logs tomorrow.

**Reply by baudneo** [reply]
If I remember correctly the older v2 main stream sends a non baseline profile-id and the substream does not send a profile-id at all. I am going to assume that is the issue here. 

Idk if motion eye allows for overriding the profile-id, I do know that the encoding for both hi-res (main stream) and low-res (substream) can be displayed properly. I had the same problem using webrtc software janus but, janus allows for overriding the profile-id (fmtp params) which in the end allowed me to stream properly.

**Reply by Bob-AOL** [reply]
So would this then primarily be a hardware, firmware, or application level issue?

I ask because I'm picking up a handful of older V2s today and don't want to build a Linux box just to facilitate the hack to discover it's as problematic with V2s as described here. The main application platform will be Blue Iris, fwiw.

**Reply by bjs-pdx** [reply]
Just curious...I can connect a V2 running the Wyze implementation of RTSP to MotionEye, but not the one used here.  So I have to assume there is something different about this version that is causing the problem rather than something unique about the hardware.  Would you agree?

**Reply by Bob-AOL** [reply]
If you're bouncing that my way, I would lean in that direction.

Just picked up the V2s and am going through them now. Three with 32gB cards for 30 bux total . Right up my alley for learning them the cheap hard way. v9.8.501 firmware in these rascals. Should I upgrade to 1002 (the latest) as the foundation for any of the other stuff? First experiment will be flashing the Wyze RTSP firmware to see where that goes while I figure out a realistic approach to compiling the mini version to get the POE and ethernet operational.

And somewhere along the line I'm also gonna hafta mow my lawn. LOL

**Reply by baudneo** [reply]
> Just curious...I can connect a V2 running the Wyze implementation of RTSP to MotionEye, but not the one used here.  So I have to assume there is something different about this version that is causing the problem rather than something unique about the hardware.  Would you agree?  

Hard to say, if you're running a lim8x box I would recommend taking a look at zoneminder as your NVR. I am a part of the zm team, I am rewriting the AI/ML for object detection.

I'd need more info on the v2cs as I assumed all v2's were the same? ZoneMinder handles the wyze stock rtsp and the wzfmini firmware perfectly fyi.

**Reply by bjs-pdx** [reply]
Running MotionEye on a Pi4 right now, just for testing so no ZM.  The V2 I'm testing is fairly new.  I can test on a PanV1 if that hardware is the same as the V2.  Otherwise, I need to pull-out one of my older V2s for testing.

Update:  Pan V1 works fine.  Installed the hack and RTSP works correctly.  Going try reloading on the V2.

**Reply by bjs-pdx** [reply]
Just noticed something unexpected.  The RTSP stream is live even when the camera shows as off in the Wyze App.  Noticed this on my Pan V2 while testing other cameras.  Is this an expected behaviour?  I know that the camera doesn't really shut off but I didn't expect live video from RTSP when the app shows the camera as off.

**Reply by gtxaspec** [reply]
we intercept the video before it's post-processed by iCamera.  The only options in the app that modify the rtsp video should be: on screen display, orientation, and night mode.  rtsp audio and video are always on

**Reply by baudneo** [reply]
> Running MotionEye on a Pi4 right now, just for testing so no ZM.  The V2 I'm testing is fairly new.  I can test on a PanV1 if that hardware is the same as the V2.  Otherwise, I need to pull-out one of my older V2s for testing.
> 
> Update:  Pan V1 works fine.  Installed the hack and RTSP works correctly.  Going try reloading on the V2.

Zm runs on pi's, I would get a usb TPU for AI/ML though.

**Comment by bjs-pdx** [comment, marked as answer]
I finally found the problem.  I have another camera from Amcrest that was grabbing the same IP address for its wireless connection, even though I had wifi disabled on the camera.  It wasn't showing in my DHCP leases, it was just grabbing the IP from memory and activating it.
